### PR TITLE
Reworking Most simple mobs hunger

### LINKED ISSUE #6257: penny's rest sprite is not used/not working properly
when penny rests, the calico sprite is shown. the rest sprite IS in the code and should work properly as it has in the past, but something seems to have broken.

when an animal is dropped, it momentarily shows the resting sprite. this is the only time when penny's proper resting sprite works.

### PR DESCRIPTION
**Pull request by Sindorman**
This PR finally reworks most of simple mobs hunger mechanics to make them not eating like crazy.

- corgies and foxes use generic formula instead of weird ass formula.

- Max nutrition values have been adjusted and tested for cats, corgies, foxes, dogs, mice. They would last 30-40 minutes depending on mob's size. Mice lasts 20 minutes. Also this rework makes these mobs have proper max values so you don't need to feed IAN 6 steaks in order to fill it up.

- simple mobs with nutrition equal or below 5% will automatically seek food.

- Rat king now adjusts it's nutrition stats according to the growth.

- Mob sizes have been changed to make more sense(since pugs were bigger than corgies, cats and foxes).

- Starving mobs will now try and steal food from people's hands instead of begging. They still beg if they are just hungry.

- Fixes #6257

- Fixes simple mobs begging food from people far away(supposed to be only adjacent)

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/commanded/guard_dog.dm [hunk 1]
```diff
@@ -29,15 +29,16 @@
...
+	mob_size = 5
 
 	response_help = "pets"
 	response_harm = "hits"
 	response_disarm = "pushes"
 
 	hunger_enabled = 1 //so you can feed your dog or something
-	autoseek_food = 0
-	beg_for_food = 0
+	autoseek_food = 1
```
Comment (Arrow768) [inline code/modules/mob/living/simple_animal/hostile/commanded/guard_dog.dm]: I´d say thats a nogo for the guard dog (autoseek_food and beg_for_food)

Comment (Sindorman) [reply code/modules/mob/living/simple_animal/hostile/commanded/guard_dog.dm]: Any dog, even guard dog would seek food when they are starving. Maybe not beg

Comment (Sindorman) [reply code/modules/mob/living/simple_animal/hostile/commanded/guard_dog.dm]: I will remake it to be turned off by default. But turn on when it is starving

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 2]
```diff
@@ -594,8 +598,25 @@ mob/living/simple_animal/bullet_act(var/obj/item/projectile/Proj)
...
-					else if(ishuman(movement_target.loc) && Adjacent(src, get_turf(movement_target)) && prob(10))
-						beg(movement_target, movement_target.loc)
+					else if(ishuman(movement_target.loc) && Adjacent(get_turf(movement_target.loc)) && (prob(10) || nutrition / max_nutrition <= 0.25))
+						var/mob/living/carbon/human/H = movement_target.loc
+						var/obj/item/weapon/reagent_containers/food/snacks/F = null
+						if(istype(H.l_hand, /obj/item/weapon/reagent_containers/food/snacks))
+							F = H.l_hand
+
+						else if(istype(H.r_hand, /obj/item/weapon/reagent_containers/food/snacks))
+							F = H.r_hand
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: There's no `if (!F) return` command here. So the following code might well be executed with `F == null`.

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 3]
```diff
@@ -594,8 +598,25 @@ mob/living/simple_animal/bullet_act(var/obj/item/projectile/Proj)
 						UnarmedAttack(movement_target)
 						if (get_turf(movement_target) == loc)
 							set_dir(pick(NORTH, SOUTH, EAST, WEST, NORTH, NORTH))//Face a random direction when eating, but mostly upwards
-					else if(ishuman(movement_target.loc) && Adjacent(src, get_turf(movement_target)) && prob(10))
-						beg(movement_target, movement_target.loc)
+					else if(ishuman(movement_target.loc) && Adjacent(get_turf(movement_target.loc)) && (prob(10) || nutrition / max_nutrition <= 0.25))
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: much like beg is a function, make fetching the food a function. Makes shit more readable.

**Pull request merged**